src/data/Auth/index.tsx

```diff
@@ -0,0 +1,61 @@
+import { auth } from 'firebase'
+import { computed, observable } from 'mobx'
+import React from 'react'
+
+const defaultSucc = (user: any) => {
+  console.log(`Successfully signed in! \n\n ${JSON.stringify(user, null, 2)}`)
+}
+
+const defaultErr = (err: any) => {
+  console.error(
+    `Could not sign in through google. \n\n ${JSON.stringify(err, null, 2)}`
+  )
+}
+
+class IAuthStore {
+  @observable user: any = null
+  @observable authPending = true
+
+  listenToAuth: any
+
+  constructor() {
+    this.listenToAuth = auth().onAuthStateChanged(
+      (user) => {
+        this.user = user
+
+        this.authPending = false
+      },
+      (err) => {
+        console.error(err)
+      }
+    )
+  }
+
+  public async doSignInWithGoogle() {
+    const provider = new auth.GoogleAuthProvider()
+    await auth().signInWithPopup(provider).then(defaultSucc, defaultErr)
+  }
+
+  public async doSignInWithEmail(e: string, p: string) {
+    await auth().signInWithEmailAndPassword(e, p).then(defaultSucc, defaultErr)
+  }
+
+  @computed
+  public get isAuthed() {
+    return !!this.user
```

Comment:
There supposed to be two `!`s here?

Reply:
@safinsingh that ensures the result is a boolean since `this.user` is a non-boolean